From 8419595352b76ccbef96b5241348a93feb3ec2d3 Mon Sep 17 00:00:00 2001
From: Harry Wentland <harry.wentland@amd.com>
Date: Thu, 6 Oct 2016 15:45:41 -0400
Subject: [PATCH 1220/1722] drm/amd/dal: Don't include underlay in max_targets

The underlay pipe is special and can't be used as a target/crtc

Change-Id: Id9127f6daab6612d736c2dab1ea6206e6f688dee
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc.c b/drivers/gpu/drm/amd/dal/dc/core/dc.c
index 794490c..76dc9ac 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc.c
@@ -604,6 +604,7 @@ void ProgramPixelDurationV(unsigned int pixelClockInKHz )
 struct dc *dc_create(const struct dc_init_data *init_params)
  {
 	struct core_dc *core_dc = dm_alloc(sizeof(*core_dc));
+	bool has_underlay = false;
 
 	if (NULL == core_dc)
 		goto alloc_fail;
@@ -617,6 +618,12 @@ struct dc *dc_create(const struct dc_init_data *init_params)
 	core_dc->public.caps.max_targets = dm_min(
 			core_dc->res_pool->pipe_count,
 			core_dc->res_pool->stream_enc_count);
+
+	/* Don't report underlay pipe as target */
+	has_underlay = core_dc->res_pool->underlay_pipe_index >= 0;
+	if (has_underlay)
+		core_dc->public.caps.max_targets--;
+
 	core_dc->public.caps.max_links = core_dc->link_count;
 	core_dc->public.caps.max_audios = core_dc->res_pool->audio_count;
 
-- 
2.7.4

